salt/grains/core.py

```diff
@@ -567,22 +568,29 @@ def _aix_memdata():
     '''
     grains = {'mem_total': 0, 'swap_total': 0}
 
-    prtconf = '/usr/sbin/prtconf 2>/dev/null'
-    for line in __salt__['cmd.run'](prtconf, python_shell=True).splitlines():
-        comps = line.split(' ')
-        if comps[0].strip() == 'Memory' and comps[1].strip() == 'Size:':
-            grains['mem_total'] = int(comps[2].strip())
-            break
+    prtconf = salt.utils.path.which('prtconf')
+    if prtconf:
+        for line in __salt__['cmd.run'](prtconf, python_shell=True).splitlines():
+            comps = line.split(' ')
+            if comps[0].strip() == 'Memory' and comps[1].strip() == 'Size:':
+                grains['mem_total'] = int(comps[2].strip())
```

Comment:
Like I already mentioned, this sort of parsing is very unreliable and fragile. You also constantly stripping things as you are unsure what is inside. My proposed version instead:
```python
comps = [x for x in line.strip().split(' ') if x]  # Or use filter() if you like
if len(comps) > 2 and 'Memory' in comps[0] and 'Size' in comps[1]:
    grains['mem_total'] = int(comps[2])

Reply:
The Solaris code from which the AIX support was derived had previously been limited to Python 2.6, hence no list comprehensions. Corrected now as per your suggestion, since base level is Python 2.7.